fastlane/lib/fastlane/actions/notarize.rb

```diff
@@ -1,13 +1,29 @@
 module Fastlane
   module Actions
     class NotarizeAction < Action
+      PASSWORD_ENV_KEY = 'FL_NOTARIZE_PASSWORD'
+
       def self.run(params)
         package_path = params[:package]
         bundle_id = params[:bundle_id]
         try_early_stapling = params[:try_early_stapling]
         print_log = params[:print_log]
         verbose = params[:verbose]
 
+        # Add the app-specific password as a temporary environment variable for
+        # altool.
+        app_specific_password_key = 'FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD'
+        password = ENV[app_specific_password_key]
+
+        message = %{Please specify an app specific password to access App Store Connect for the notarization process.
+          You can do so via the #{app_specific_password_key} environment variable.
+          More information at https://docs.fastlane.tools/best-practices/continuous-integration/#application-specific-passwords.
+        }
+
+        UI.user_error!(message) if password.nil? || password.empty?
```

Comment:
The password empty is a bit of an extra, but I think it's a reasonable check to make.